google-cloud-bigtable/lib/google/cloud/bigtable/instance/cluster_map.rb

```diff
@@ -45,24 +48,61 @@ def initialize value = {}
+          # @param nodes [Integer] Number of nodes for the cluster. When creating an instance of type `:DEVELOPMENT`,
+          #   `nodes` must not be set.
+          # @param storage_type [Symbol] The type of storage used by this cluster to serve its parent instance's tables,
+          #   unless explicitly overridden. Valid values are:
+          #
+          #   * `:SSD` - Flash (SSD) storage should be used.
+          #   * `:HDD` - Magnetic drive (HDD) storage should be used.
+          #
+          #   If not set then default will set to `:STORAGE_TYPE_UNSPECIFIED`.
+          # @param kms_key [String] The full name of a Cloud KMS encryption key for a CMEK-protected cluster, in the
+          #   format `projects/{key_project_id}/locations/{location}/keyRings/{ring_name}/cryptoKeys/{key_name}`.
+          #
+          #   The requirements for this key are:
+          #
+          #   1. The Cloud Bigtable service account associated with the project that contains this cluster must be
+          #      granted the `cloudkms.cryptoKeyEncrypterDecrypter` role on the CMEK key.
+          #   2. Only regional keys can be used and the region of the CMEK key must match the region of the cluster.
+          #   3. All clusters within an instance must use the same CMEK key.
+          #
+          # @example
+          #   require "google/cloud/bigtable"
+          #
+          #   bigtable = Google::Cloud::Bigtable.new
+          #
+          #   job = bigtable.create_instance("my-instance") do |clusters|
+          #     clusters.add("test-cluster", "us-east1-b", nodes: 3, storage_type: :SSD)
+          #   end
+          #
+          #   job.wait_until_done!
+          #
+          # @example With a Cloud KMS encryption key name for a CMEK-protected cluster:
+          #   require "google/cloud/bigtable"
+          #
+          #   bigtable = Google::Cloud::Bigtable.new
+          #
+          #   kms_key_name = "projects/a/locations/b/keyRings/c/cryptoKeys/d"
+          #   job = bigtable.create_instance("my-instance") do |clusters|
+          #     clusters.add("test-cluster", "us-east1-b", kms_key: kms_key_name)
+          #   end
           #
-          #   If not set then default will set to `:STORAGE_TYPE_UNSPECIFIED`
+          #   job.wait_until_done!
           #
-          def add name, location, nodes: nil, storage_type: nil
+          def add name, location, nodes: nil, storage_type: nil, kms_key: nil
+            if kms_key
+              encryption_config = Google::Cloud::Bigtable::Admin::V2::Cluster::EncryptionConfig.new(
+                kms_key_name: kms_key
+              )
+            end
             attrs = {
               serve_nodes:          nodes,
               location:             location,
-              default_storage_type: storage_type
+              default_storage_type: storage_type,
+              encryption_config: encryption_config
```

Comment:
Maybe keep using the aligned hashes if the rest of the hash is still using it.

Reply:
Fixed, thanks!